library/x509_crl.c

```diff
@@ -294,7 +294,7 @@ int mbedtls_x509_crl_parse_der( mbedtls_x509_crl *chain,
     /*
      * Copy raw DER-encoded CRL
      */
-    if( ( p = mbedtls_calloc( 1, buflen ) ) == NULL )
+    if( buflen != 0 && ( p = mbedtls_calloc( 1, buflen ) ) == NULL )
```

Comment:
to make it more readable, please surround the statement `( p = mbedtls_calloc( 1, buflen ) ) == NULL` with parentheses